hooks/repo.go

```diff
@@ -14,6 +15,7 @@ type HookRepository interface {
 	GetRawBidResponseHook(id string) (hookstage.RawBidResponse, bool)
 	GetAllProcessedBidResponsesHook(id string) (hookstage.AllProcessedBidResponses, bool)
 	GetAuctionResponseHook(id string) (hookstage.AuctionResponse, bool)
+	GetAllModuleStageNames() map[string][]string
```

Comment:
I propose not extending the `HookRepository` interface with this method. Instead, you can extend the `modules.Builder#Build` function to return a list of module stages in addition to the `HookRepository`, ex: `Build(cfg config.Modules, client *http.Client) (hooks.HookRepository, map[string][]string, error)`.

The trade-off is that the logic for casting hook interfaces will need to be repeated to collect module stages, but I think a narrower interface is preferred over repeating yourself, at least in this case.